hibernate-core/src/main/java/org/hibernate/tuple/entity/EntityMetamodel.java

```diff
@@ -97,7 +97,7 @@
 
 	private final Map<String, Integer> propertyIndexes = new HashMap<>();
 	private final boolean hasCollections;
-	private final boolean hasMutableProperties;
+	private final int[] mutablePropertiesIndexes;
```

Comment:
You think we could use BitSet(s) to track such things?  Probably not worth our time here, but I'd like to take that in consideration in ORM 6.

Reply:
Yeah, a `BitSet` is a good candidate for something like that.